### PR TITLE
[Core] Fix comment about escaping characters in AppendPath()

### PR DESCRIPTION
**Pull request by Mohit-Chakraborty**

### CODE REVIEW

**Comment by annelo-msft** [comment]
Curious: what motivated this?

**Review by annelo-msft — APPROVED** [review]

**Comment by Mohit-Chakraborty** [comment]
> Curious: what motivated this?

I am in the middle of debugging a service error for the ACR SDK. I chanced upon this comment while looking at the code and it was relevant to me because we have a ':' in the path.

File: sdk/core/Azure.Core/src/RequestUriBuilder.cs [hunk 1]
```diff
@@ -199,7 +199,7 @@ public void AppendQuery(string name, string value, bool escapeValue)
 
         /// <summary>
         /// Escapes and appends the <paramref name="value"/> to <see cref="Path"/> without adding path separator.
-        /// Path segments and any other characters will be escaped, e.g. "/" will be escaped as "%3a".
+        /// Path segments and any other characters will be escaped, e.g. ":" will be escaped as "%3a".
```
[the remark this answers is not in the document]
Comment (Mohit-Chakraborty) [reply sdk/core/Azure.Core/src/RequestUriBuilder.cs:202]: I did it so we get different common examples out with different doc comments. Using the same example would look consistent but add no additional value. That was my thinking. If it looks weird, we can use the same example in both cases.

[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/core/Azure.Core/src/RequestUriBuilder.cs:202]: I don't think it adds much more value, as there are many other reserved characters so this isn't an exhaustive set of examples. I'd suggest either linking to some authoritative document on percent encoding or just sticking with the single example in both places (or doing both).

[the remark this answers is not in the document]
Comment (Mohit-Chakraborty) [reply sdk/core/Azure.Core/src/RequestUriBuilder.cs:202]: Done.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**